addons/web/static/src/core/select_menu/select_menu.js

```diff
@@ -96,7 +96,8 @@ export class SelectMenu extends Component {
             }
             if (
                 this.props.choices.length !== nextProps.choices.length ||
-                this.props.groups.length !== nextProps.groups.length
+                this.props.groups.length !== nextProps.groups.length ||
+                JSON.stringify(this.props.groups) !== JSON.stringify(nextProps.groups)
```

Comment:
Don't we need to do the same kind of verification for "choices" then?

And also include a test case for this in "select_menu_test.js"?

Hereby summoning @Iucapad for a second check.